ts/session/apis/snode_api/retrieveRequest.ts

```diff
@@ -123,14 +124,12 @@ async function retrieveNextMessages(
   );
   // let exceptions bubble up
   // no retry for this one as this a call we do every few seconds while polling for messages
-
-  const results = await doSnodeBatchRequest(
-    retrieveRequestsParams,
-    targetNode,
-    4000,
-    associatedWith
-  );
-  if (!results || !results.length) {
+  const timeOutMs = 4 * 1000;
+  const results = await Promise.race([
```

Comment:
This seems magical first time I have heard of Promise.race. Any particular reason for 4 seconds?